tests/integration/test_bundle.py

```diff
@@ -0,0 +1,147 @@
+import asyncio
+import json
+import logging
+
+from pathlib import Path
+
+import aiohttp
+import lightkube
+import lightkube.codecs
+import pytest
+import requests
+import tenacity
+import yaml
+from pytest_operator.plugin import OpsTest
+
+ISVC = lightkube.generic_resource.create_namespaced_resource(
+    group="serving.kserve.io",
+    version="v1beta1",
+    kind="InferenceService",
+    plural="inferenceservices",
+    verbs=None,
+)
+BUNDLE_PATH = "./bundle/bundle.yaml"
+SKLEARN_INF_SVC_YAML = yaml.safe_load(
+    Path("./tests/integration/sklearn-iris.yaml").read_text()
+)
+SKLEARN_INF_SVC_OBJECT = lightkube.codecs.load_all_yaml(
+    yaml.dump(SKLEARN_INF_SVC_YAML)
+)[0]
+SKLEARN_INF_SVC_NAME = SKLEARN_INF_SVC_OBJECT.metadata.name
+
+
+logger = logging.getLogger(__name__)
+
+
+@pytest.fixture(scope="session")
+def lightkube_client() -> lightkube.Client:
+    client = lightkube.Client(field_manager="autoscaling-bundle")
+    return client
+
+
+async def cli_deploy_bundle(ops_test: OpsTest, bundle_path: str):
+    """Deploy bundle from charmhub or from file."""
+    run_args = [
+        "juju",
+        "deploy",
+        "--trust",
+        "-m",
+        ops_test.model_full_name,
+        bundle_path,
+    ]
+
+    retcode, stdout, stderr = await ops_test.run(*run_args)
+    assert retcode == 0, f"Deploy failed: {(stderr or stdout).strip()}"
+    logger.info(stdout)
+
+
+@pytest.mark.abort_on_fail
+async def test_build_and_deploy(ops_test: OpsTest):
+    """Deploy the autoscaling-model-server bundle."""
+    await cli_deploy_bundle(ops_test, bundle_path=BUNDLE_PATH)
+    # Configure knative-serving with Istio information
+    await ops_test.model.applications["knative-serving"].set_config(
+        {
+            "istio.gateway.name": ISTIO_GATEWAY_NAME,
+            "istio.gateway.namespace": ops_test.model.name,
+        }
+    )
```

Comment:
Shouldn't this be already set in the bundle.yaml file?

Reply:
Actually, I see that at least one configuration is namespace related.

Reply:
Yes, because the namespace may change, this will be done by users. That's how knative-serving was design from the beginning. The alternative to this would be to use the `gateway-info` relation instead.

Reply:
But would that would need to push appropriate changes to knative-serving charm, right?

Reply:
> But would that would need to push appropriate changes to knative-serving charm, right?

You mean about the relation? If so, yes, if we want to use the relation, we will need to push changes to the knative-serving charm.

Reply:
Ok we can open a discussion in the knative charms repository about it, WDYT?

Reply:
There is an issue already https://github.com/canonical/knative-operators/issues/28